sycl/include/sycl/accessor.hpp

```diff
@@ -1145,6 +1145,33 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
       : impl({}, detail::InitializedVal<AdjustedDim, range>::template get<0>(),
              detail::InitializedVal<AdjustedDim, range>::template get<0>()) {}
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_, int Dimensions_, access::mode AccessMode_,
+            access::target AccessTarget_, access::placeholder IsPlaceholder_,
+            typename PropertyListT_>
+  accessor(const accessor<DataT_, Dimensions_, AccessMode_, AccessTarget_,
+                          IsPlaceholder_, PropertyListT_> &other,
+           std::enable_if_t<(AccessMode_ == access_mode::read) &&
```

Comment:
```suggestion
  template <typename DataT_>
  accessor(const accessor<DataT_, Dimensions, AccessMode, AccessTarget,
                          IsPlaceholder, PropertyListT> &other,
           std::enable_if_t<(AccessMode == access_mode::read) &&
```
To my knowledge they should have the same template arguments, apart from the data type. That said, it should only be enabled if the access mode is read-only for this class.

Reply:
thanks @steffenlarsen, you are right. Done.